File: rh_mobile_app/lib/features/demandes_admin/data/type_conge.dart
/// Catalogue fermé M01 — types de congé (story 7).
class TypeCongeOption {
  const TypeCongeOption({required this.code, required this.label});

  final String code;
  final String label;
}

const List<TypeCongeOption> kTypesConge = [
  TypeCongeOption(code: 'ANNUEL', label: 'Congé annuel'),
  TypeCongeOption(code: 'MALADIE', label: 'Congé maladie'),
  TypeCongeOption(code: 'MATERNITE', label: 'Congé maternité'),
  TypeCongeOption(code: 'SANS_SOLDE', label: 'Congé sans solde'),
  TypeCongeOption(code: 'AUTRE', label: 'Autre'),
];

bool typeCongeExigePieceJointe(String code) =>
    code == 'MALADIE' || code == 'MATERNITE';

String libelleTypeConge(String code) {
  for (final t in kTypesConge) {
    if (t.code == code) return t.label;
  }
  return code;
}

File: svc-referentiel-rh/src/main/java/com/hr/referentiel/domain/TypeConge.java
package com.hr.referentiel.domain;

import java.util.Locale;

/**
 * Catalogue fermé des types de congé ({@code contenu.type_conge}) — M01 § 6.2.
 * PJ / certificat obligatoire uniquement pour {@link #MALADIE} et {@link #MATERNITE}.
 */
public enum TypeConge {

	ANNUEL,
	MALADIE,
	MATERNITE,
	SANS_SOLDE,
	AUTRE;

	public static final String VALEURS_ACCEPTEES =
			"ANNUEL, MALADIE, MATERNITE, SANS_SOLDE, AUTRE";

	public static final String MESSAGE_INVALIDE =
			"type_conge invalide : " + VALEURS_ACCEPTEES + ".";

	public static final String MESSAGE_CERTIFICAT_OBLIGATOIRE =
			"Un certificat / pièce jointe est obligatoire pour un congé MALADIE ou MATERNITE.";

	/** {@code true} si une pièce jointe (certificat médical / maternité) est exigée. */
	public boolean exigePieceJointe() {
		return this == MALADIE || this == MATERNITE;
	}

	/**
	 * Parse une valeur catalogue. {@code null} / blank → {@link IllegalArgumentException}
	 * (le champ {@code type_conge} est déjà obligatoire en amont).
	 */
	public static TypeConge parse(String raw) {
		if (raw == null || raw.isBlank()) {
			throw new IllegalArgumentException(MESSAGE_INVALIDE);
		}
		try {
			return valueOf(raw.trim().toUpperCase(Locale.ROOT));
		} catch (IllegalArgumentException e) {
			throw new IllegalArgumentException(MESSAGE_INVALIDE);
		}
	}
}
